contracts/modules/royalty/policies/IpRoyaltyVault.sol

```diff
@@ -133,132 +132,34 @@ contract IpRoyaltyVault is IIpRoyaltyVault, ERC20SnapshotUpgradeable, Reentrancy
         _updateVaultBalance(token, amount);
     }
 
-    /// @notice Snapshots the claimable revenue and royalty token amounts
-    /// @return The snapshot id
-    function snapshot() external whenNotPaused returns (uint256) {
-        IpRoyaltyVaultStorage storage $ = _getIpRoyaltyVaultStorage();
-
-        if (block.timestamp - $.lastSnapshotTimestamp < IVaultController(address(ROYALTY_MODULE)).snapshotInterval())
-            revert Errors.IpRoyaltyVault__InsufficientTimeElapsedSinceLastSnapshot();
-
-        uint256 snapshotId = _snapshot();
-        $.lastSnapshotTimestamp = uint40(block.timestamp);
-
-        uint256 noRevenueCounter;
-        address[] memory tokenList = $.tokens.values();
-        for (uint256 i = 0; i < tokenList.length; i++) {
-            if (IERC20Upgradeable(tokenList[i]).balanceOf(address(this)) == 0) {
-                $.tokens.remove(tokenList[i]);
```

Comment:
Why do we need to track revenue tokens with balance zero? What breaks if we remove the token from the token list?

Can't we have something along the lines of

```
if (IERC20Upgradeable(tokenList[i]).balanceOf(address(this)) == 0) {
                $.tokens.remove(tokenList[i]);
```

inside the function `_claimRevenueOnBehalf` ?

Reply:
We need to keep the token in the token list even when its balance is zero because removing it could prevent us from properly adjusting a user’s `revenueDebt` for that token when they transfer RTs to another user, as the token would no longer be listed. Since the token’s balance can be increased later, the user’s claimable revenue for that token might become inaccurate if it was previously removed from the list.